operator-common/src/main/java/io/strimzi/operator/common/http/HealthCheckAndMetricsServer.java

```diff
@@ -115,62 +115,66 @@ public void stop() {
     /**
      * Handler responsible for the liveness check
      */
-    class HealthyHandler extends AbstractHandler {
+    class HealthyHandler extends Handler.Abstract {
         @Override
-        public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
-            response.setContentType("application/json");
+        public boolean handle(Request request, Response response, Callback callback) throws Exception {
+            response.getHeaders().put(HttpHeader.CONTENT_TYPE, "application/json; charset=UTF-8");
 
             if (liveness.isAlive()) {
                 response.setStatus(HttpServletResponse.SC_OK);
-                response.getWriter().println("{\"status\": \"ok\"}");
+                response.write(true, StandardCharsets.UTF_8.encode("{\"status\": \"ok\"}"), callback);
             } else {
                 response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-                response.getWriter().println("{\"status\": \"not-ok\"}");
+                response.write(true, StandardCharsets.UTF_8.encode("{\"status\": \"not-ok\"}"), callback);
             }
+
             LOGGER.debug("Responding {} to GET /healthy", response.getStatus());
-            baseRequest.setHandled(true);
+
+            return true;
         }
     }
 
     /**
      * Handler responsible for the readiness check
      */
-    class ReadyHandler extends AbstractHandler {
+    class ReadyHandler extends Handler.Abstract {
         @Override
-        public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
-            response.setContentType("application/json");
+        public boolean handle(Request request, Response response, Callback callback) throws Exception {
+            response.getHeaders().put(HttpHeader.CONTENT_TYPE, "application/json; charset=UTF-8");
```

Comment:
AFAIK, UTF-8 is the standard for JSON and adding the charset here is option when using application/json. Did you have any troubles or reasons to add it anyway?